packages/fxa-payments-server/src/routes/Subscriptions/index.tsx

```diff
@@ -266,19 +265,13 @@ const subscriptionForId = (
   subscriptionId: string,
   subscriptions: SubscriptionsFetchState
 ): Subscription | null => {
-  if (subscriptions.result === null) {
-    return null;
-  }
-  return subscriptions.result.filter(
+  return (subscriptions.result as Subscription[]).filter(
     subscription => subscription.subscriptionId === subscriptionId
   )[0];
 };
 
 const planForId = (planId: string, plans: PlansFetchState): Plan | null => {
```

Comment:
This ` | null` return type and the above one can be removed now.

Reply:
Here, though, it *is* still possible for a null value - there's an edge case issue where if subhub & fxa subscription data for a user are out of sync, one or the other can be missing a particular subscription and we have an error dialog for that

Reply:
I'm a little confused.  Simply looking at the return statement below, it appears that a null is not possible since the first element of an array is returned.  I suppose if the result of the `filter` is an empty array, then `[0]` of it is undefined.  (I trust the TS compiler; just confused from reading the code.)